lib/video_info/providers/youtube_api.rb

```diff
@@ -83,8 +83,7 @@ def _channel_api_url(channel_id)
 
       def _channel_info
         channel_url = _channel_api_url(_video_snippet['channelId'])
-        @channel_data ||= open(channel_url)
-        JSON.load(@channel_data.read)
+        @channel_data ||=  JSON.load(open(channel_url).read)
```

Comment:
The key was to not call .read twice.  But we may as wlel just store the JSON so that work only needs to be done once.

Reply:
Looks good, that was indeed a bad implementation. 👍